src/Jobs/DeleteDomains.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Stancl\Tenancy\Contracts\TenantWithDatabase;
+use Stancl\Tenancy\Database\Models\Domain;
+use Stancl\Tenancy\Events\DatabaseDeleted;
+use Stancl\Tenancy\Events\DeletingDatabase;
+use Stancl\Tenancy\Events\DeletingDomain;
+use Stancl\Tenancy\Events\DomainDeleted;
+
+class DeleteDomains implements ShouldQueue
```

Comment:
One last change — why does this implement `ShouldQueue`? I don't think we want to queue the individual jobs in the pipeline.

Reply:
> why does this implement ShouldQueue? I don't think we want to queue the individual jobs in the pipeline.

I mainly got motivation from the DeleteDatabase job. https://github.com/archtechx/tenancy/blob/3.x/src/Jobs/DeleteDatabase.php. DeleteDomains class is an exact copy of the DeleteDatabase class, and both classes are being used in the same way I mean in JobPipeline https://github.com/archtechx/tenancy/blob/3.x/assets/TenancyServiceProvider.stub.php#L45

Reply:
I see. Database is deleted *after* the tenant is deleted. Domains are deleted *before*. So it needs to be synchronous.

I'm not sure how the JobPipeline interprets jobs having that interface though. Could you check that? Use that repo's readme and source code.

Reply:
> After looking into the Pipeline source code, I think it does not matter if the Job class has shouldQueue interface or not.

I see, you're right. Anyway, here we'll remove it because it's a `deleting` event listener 👍🏻